src/AssemblerHttpServer-Alignments.cpp

```diff
@@ -1520,23 +1520,29 @@ void Assembler::computeAllAlignmentsThreadFunction(size_t threadId)
                 getMarkersSortedByKmerId(orientedReadId1, markersSortedByKmerId[1]);
 
                 // Compute the alignment.
-                if (method == 0) {
-                    const bool debug = false;
-                    alignOrientedReads(
-                        markersSortedByKmerId,
-                        maxSkip, maxDrift, maxMarkerFrequency, debug, graph, alignment, alignmentInfo);
-                } else if (method == 1) {
-                    alignOrientedReads1(
-                        orientedReadId0, orientedReadId1,
-                        matchScore, mismatchScore, gapScore, alignment, alignmentInfo);
-                } else if (method == 3) {
-                    alignOrientedReads3(
-                        orientedReadId0, orientedReadId1,
-                        matchScore, mismatchScore, gapScore,
-                        downsamplingFactor, bandExtend, maxBand,
-                        alignment, alignmentInfo);
-                } else {
-                    SHASTA_ASSERT(0);
+                try {
+                    if (method == 0) {
+                        const bool debug = false;
+                        alignOrientedReads(
+                            markersSortedByKmerId,
+                            maxSkip, maxDrift, maxMarkerFrequency, debug, graph, alignment, alignmentInfo);
+                    } else if (method == 1) {
+                        alignOrientedReads1(
+                            orientedReadId0, orientedReadId1,
+                            matchScore, mismatchScore, gapScore, alignment, alignmentInfo);
+                    } else if (method == 3) {
+                        alignOrientedReads3(
+                            orientedReadId0, orientedReadId1,
+                            matchScore, mismatchScore, gapScore,
+                            downsamplingFactor, bandExtend, maxBand,
+                            alignment, alignmentInfo);
+                    } else {
+                        SHASTA_ASSERT(0);
+                    }
+                } catch (const std::exception& e) {
```

Comment:
Also add a catch block for a non-`std` exception (`catch(...)`), like in `computeAlignments`? Just in case SeqAn throws a custom exception class that is not derived from `std::exception`.